### ISSUE TITLE
Slate to Lexical converter type issue and improvements

### ISSUE THREAD

**Issue by cbratschi** [payloadcms/payload#4485]
### Link to reproduction

private repo

### Describe the Bug

The link converter uses a wrong type:

https://github.com/payloadcms/payload/blob/main/packages/richtext-lexical/src/field/features/Link/nodes/LinkNode.ts#L30

https://github.com/payloadcms/payload/blob/a2dac605e52c06b40b33f6d1af9127a257faa8d0/packages/richtext-lexical/src/field/features/migrations/SlateToLexical/converter/converters/link.ts#L17

Where value is passed as a string:

```
{ value: '64dde6cae010f20851f30427', relationTo: 'pages' }
{ value: '64d12043eae97b8250e87a36', relationTo: 'media' }
```

In this case it has no id field.

### To Reproduce

Use Slate link in editor and try to iterate the converted Lexical data. The following cast is needed:

```JS
const documentId = linkNode.fields.doc?.value as string | undefined;
```

### Payload Version

2.4.0

### Adapters and Plugins

db-mongodb, bundler-webpack, live-preview

**Comment by cbratschi** [comment]
Here the improved link converted which supports custom fields:

```JS
/**
 * Modified Slate to Lexical link converter which also copies custom field values.
 */
export const SlateLinkConverter: SlateNodeConverter = {
    converter({ converters, slateNode }) {
        return {
            children: convertSlateNodesToLexical({
                canContainParagraphs: false,
                converters,
                parentNodeType: 'link',
                slateNodes: slateNode.children || [],
            }),
            direction: 'ltr',
            fields: {
                //Note: added this line
                ...slateNode.fields || {},
                doc: slateNode.doc || undefined,
                linkType: slateNode.linkType || 'custom',
                newTab: slateNode.newTab || false,
                url: slateNode.url || undefined,
            },
            format: '',
            indent: 0,
            type: 'link',
            version: 1,
        } as const as SerializedLinkNode;
    },
    nodeTypes: ['link']
};
```

One line was added.

**Comment by cbratschi** [comment]
And the blockquote conversion was missing:

```JS
/**
 * Convert blockquotes from Slate to Lexical.
 */
const SlateBlockquoteConverter: SlateNodeConverter = {
    converter({ converters, slateNode }) {
        return {
            children: convertSlateNodesToLexical({
                canContainParagraphs: false,
                converters,
                parentNodeType: 'quote',
                slateNodes: slateNode.children || []
            }),
            direction: 'ltr',
            format: '',
            indent: 0,
            type: 'quote',
            version: 1,
        } as const as SerializedQuoteNode;
    },
    nodeTypes: ['blockquote']
};
```

**Comment by cbratschi** [comment]
@AlessioGr one bug two code improvements for you

**Comment by AlessioGr** [comment]
Thank you for the code snippets! This has been merged in and will be available in the next version :)

**Comment by cbratschi** [comment]
@AlessioGr thanks but the original type issue I reported at the top was not fixed.

**Comment by AlessioGr** [comment]
@cbratschi will be fixed in #4504. It changes the data format to match the one used in Slate.
